File: api/models/account_money_extend.py
from extensions.ext_database import db

from .types import StringUUID


class AccountMoneyExtend(db.Model):
    __tablename__ = "account_money_extend"
    __table_args__ = (
        db.PrimaryKeyConstraint("id", name="account_money_pkey"),
        db.Index("idx_account_money_account_id", "account_id"),
    )

    id = db.Column(StringUUID, server_default=db.text("uuid_generate_v4()"))
    account_id = db.Column(StringUUID, nullable=False)
    total_quota = db.Column(db.Numeric(16, 7))
    used_quota = db.Column(db.Numeric(16, 7))
    created_at = db.Column(db.DateTime, nullable=False, server_default=db.text("CURRENT_TIMESTAMP(0)"))
    updated_at = db.Column(db.DateTime, nullable=False, server_default=db.text("CURRENT_TIMESTAMP(0)"))


class AccountLayoverRecordExtend(db.Model):
    __tablename__ = "account_layover_record_extend"
    __table_args__ = (
        db.PrimaryKeyConstraint("id", name="account_layover_record_extend_pkey"),
        db.Index("idx_account_layover_record_account_id", "account_id"),
        db.Index("idx_account_layover_record_forwarding_id", "forwarding_id"),
    )

    id = db.Column(StringUUID, server_default=db.text("uuid_generate_v4()"))
    account_id = db.Column(StringUUID, nullable=False)
    forwarding_id = db.Column(StringUUID, nullable=False)
    money = db.Column(db.Numeric(16, 7))
    info = db.Column(db.JSON, default={})
    created_at = db.Column(db.DateTime, nullable=False, server_default=db.text("CURRENT_TIMESTAMP(0)"))

File: admin/server/model/gaia/account_money_extend.go
// 自动生成模板AccountMoneyExtend
package gaia

import (
	"github.com/gofrs/uuid/v5"
	"time"
)

// accountMoneyExtend表 结构体  AccountMoneyExtend
type AccountMoneyExtend struct {
	Id         *string    `json:"id" form:"id" gorm:"primarykey;column:id;comment:;"`                        //id字段
	AccountId  uuid.UUID  `json:"accountId" form:"accountId" gorm:"uniqueIndex;column:account_id;comment:;"` //accountId字段
	TotalQuota float64    `json:"totalQuota" form:"totalQuota" gorm:"column:total_quota;comment:;"`          //totalQuota字段
	UsedQuota  float64    `json:"usedQuota" form:"usedQuota" gorm:"column:used_quota;comment:;"`             //usedQuota字段
	CreatedAt  *time.Time `json:"createdAt" form:"createdAt" gorm:"column:created_at;comment:;size:6;"`      //createdAt字段
	UpdatedAt  *time.Time `json:"updatedAt" form:"updatedAt" gorm:"column:updated_at;comment:;size:6;"`      //updatedAt字段
}

// TableName accountMoneyExtend表 AccountMoneyExtend自定义表名 account_money_extend
func (AccountMoneyExtend) TableName() string {
	return "account_money_extend"
}
